modules/component-store/src/component-store.ts

```diff
@@ -149,58 +155,71 @@ export class ComponentStore<T extends object> implements OnDestroy {
   /**
    * Creates a selector.
    *
-   * This supports chaining up to 4 selectors. More could be added as needed.
+   * This supports combining up to 4 selectors. More could be added as needed.
    *
    * @param projector A pure projection function that takes the current state and
    *   returns some new slice/projection of that state.
+   * @param config SelectConfig that changes the behavoir of selector, including
+   *   the debouncing of the values until the state is settled.
    * @return An observable of the projector results.
    */
-  select<R>(projector: (s: T) => R): Observable<R>;
-  select<R, S1>(s1: Observable<S1>, projector: (s1: S1) => R): Observable<R>;
+  select<R>(projector: (s: T) => R, config?: SelectConfig): Observable<R>;
+  select<R, S1>(
+    s1: Observable<S1>,
+    projector: (s1: S1) => R,
+    config?: SelectConfig
+  ): Observable<R>;
   select<R, S1, S2>(
     s1: Observable<S1>,
     s2: Observable<S2>,
-    projector: (s1: S1, s2: S2) => R
+    projector: (s1: S1, s2: S2) => R,
+    config?: SelectConfig
   ): Observable<R>;
   select<R, S1, S2, S3>(
     s1: Observable<S1>,
     s2: Observable<S2>,
     s3: Observable<S3>,
-    projector: (s1: S1, s2: S2, s3: S3) => R
+    projector: (s1: S1, s2: S2, s3: S3) => R,
+    config?: SelectConfig
   ): Observable<R>;
   select<R, S1, S2, S3, S4>(
     s1: Observable<S1>,
     s2: Observable<S2>,
     s3: Observable<S3>,
     s4: Observable<S4>,
-    projector: (s1: S1, s2: S2, s3: S3, s4: S4) => R
+    projector: (s1: S1, s2: S2, s3: S3, s4: S4) => R,
+    config?: SelectConfig
   ): Observable<R>;
-  select<R>(...args: any[]): Observable<R> {
-    let observable$: Observable<R>;
-    // project is always the last argument, so `pop` it from args.
-    const projector: (...args: any[]) => R = args.pop();
-    if (args.length === 0) {
-      // If projector was the only argument then we'll use map operator.
-      observable$ = this.stateSubject$.pipe(debounceSync(), map(projector));
+  select<
+    O extends Array<Observable<unknown> | SelectConfig | ProjectorFn>,
+    R,
+    ProjectorFn = (...a: unknown[]) => R
+  >(...args: O): Observable<R> {
+    const { observables, projector, config } = processSelectorArgs(args);
```

Comment:
I like that the logic of `processSelectorArgs` is extracted 👍

Reply:
Thanks Tim 😊